src/providers/Watcher/index.tsx

```diff
@@ -0,0 +1,310 @@
+import {
+  AccountResponseType,
+  GetChannelsResponseType,
+  GetInfoResponseType,
+} from '@hoprnet/hopr-sdk';
+import { useEffect } from 'react';
+import { useAppDispatch, useAppSelector } from '../../store';
+import { appActions } from '../../store/slices/app';
+import { nodeActionsAsync } from '../../store/slices/node';
+import { formatEther } from 'viem';
+
+const FETCH_CHANNELS_INTERVAL = 10000;
+const FETCH_NODE_INTERVAL = 5000;
+
+const Watcher = () => {
+  const dispatch = useAppDispatch();
+  const { apiEndpoint, apiToken } = useAppSelector(
+    (store) => store.auth.loginData
+  );
+  const { connected } = useAppSelector((store) => store.auth.status);
+  const messages = useAppSelector((store) => store.node.messages);
+
+  // previous states to compare new states with
+  let prevChannels: GetChannelsResponseType | null;
+  let prevNodeInfo: GetInfoResponseType | null;
+  let prevLoginData: {
+    apiEndpoint: string;
+    apiToken: string;
+  } | null;
+  let prevNodeFunds: AccountResponseType | null;
+  let prevMessages: {
+    createdAt: number;
+    seen: boolean;
+    body: string;
+    challenge?: string | undefined;
+  }[] = [];
+  // booleans to stop interval from running in parallel
+  let watchChannelsIntervalIsRunning = false;
+  let watchNodeFundsIntervalIsRunning = false;
+  let watchNodeInfoIntervalIsRunning = false;
+
+  useEffect(() => {
+    // reset state on every change of node
+    if (
+      prevLoginData?.apiEndpoint !== apiEndpoint ||
+      prevLoginData.apiToken !== apiToken
```

Comment:
That might backfire if someone will change apiToken for of the node and re-log.
Since we only work on notifications on the Front-End side, let's leave it for now. Maybe we will get BE based feature like that